### PR TITLE
chore(infra): Improving local setup generation

### PR DESCRIPTION
**Pull request by pshomov**
# ...

Improving the local setup generation to create .env file per service with all environment info from dev - environment variables and secrets. 

```
> yarn infra render-local-env  --service=application-system-form 
{
  services: {
    'application-system-form': '(source /Users/petar/src/andes/customers/si/island.is/.env.application-system-form && yarn start application-system-form)',
    'services-documents': '(source /Users/petar/src/andes/customers/si/island.is/.env.services-documents && yarn start services-documents)',
    'endorsement-system-api': '(source /Users/petar/src/andes/customers/si/island.is/.env.endorsement-system-api && yarn start endorsement-system-api)',
    'application-system-api': '(source /Users/petar/src/andes/customers/si/island.is/.env.application-system-api && yarn start application-system-api)',
    'icelandic-names-registry-backend': '(source /Users/petar/src/andes/customers/si/island.is/.env.icelandic-names-registry-backend && yarn start icelandic-names-registry-backend)',
    'air-discount-scheme-backend': '(source /Users/petar/src/andes/customers/si/island.is/.env.air-discount-scheme-backend && yarn start air-discount-scheme-backend)',
    'service-portal-api': '(source /Users/petar/src/andes/customers/si/island.is/.env.services-user-profile && yarn start services-user-profile)',
    'services-sessions': '(source /Users/petar/src/andes/customers/si/island.is/.env.services-sessions && yarn start services-sessions)',
    api: '(source /Users/petar/src/andes/customers/si/island.is/.env.api && yarn start api)'
  },
  mocks: 'docker run -it --rm -p 2525:2525 -p 9388:9388 -p 9372:9372 -v /Users/petar/src/andes/customers/si/island.is/infra/mountebank-imposter-config.json:/app/default.json docker.io/bbyars/mountebank:2.8.1 start --configfile=/app/default.json'
}
```
## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: infra/src/dsl/value-files-generators/local-setup.spec.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ describe('Local setup', () => {
         '..',
         '..',
         '..',
-      )}/mountebank-imposter-config.json:/app/default.json bbyars/mountebank:2.8.1 start --configfile=/app/default.json`,
+      )}/mountebank-imposter-config.json:/app/default.json docker.io/bbyars/mountebank:2.8.1 start --configfile=/app/default.json`,
```
Comment (busla) [inline infra/src/dsl/value-files-generators/local-setup.spec.ts:55]: should we maybe retag this and push to ecr shared?

File: infra/src/dsl/value-files-generators/local-setup.ts [hunk 2]
```diff
@@ -118,10 +146,13 @@ export const getLocalrunValueFile = async (
     .map((port) => `-p ${port}:${port}`)
     .join(
       ' ',
-    )} -v ${process.cwd()}/${defaultMountebankConfig}:/app/default.json bbyars/mountebank:2.8.1 start --configfile=/app/default.json`
+    )} -v ${process.cwd()}/${defaultMountebankConfig}:/app/default.json docker.io/bbyars/mountebank:2.8.1 start --configfile=/app/default.json`
 
   return {
-    services: { ...dockerComposeServices },
+    services: Object.entries(dockerComposeServices).reduce(
+      (acc, [name, service]) => ({ ...acc, [name]: service.command![0] }),
```
Comment (busla) [inline infra/src/dsl/value-files-generators/local-setup.ts:153]: should we be asserting `service.command![0]`

**Review by busla — CHANGES_REQUESTED** [review]

File: infra/src/dsl/value-files-generators/local-setup.spec.ts (hunk 1 above)
Comment (pshomov) [reply infra/src/dsl/value-files-generators/local-setup.spec.ts:55]: For local dev needs it might be too much trouble?

File: infra/src/dsl/value-files-generators/local-setup.ts (hunk 2 above)
Comment (pshomov) [reply infra/src/dsl/value-files-generators/local-setup.ts]: Excellent point, fixed.

**Review by eirikurn — COMMENTED** [review]
It's not clear to me how this works, what's different from our current `.envrc` system and how we use this. Can you write up a draft of a dev guide for this?

File: infra/src/dsl/value-files-generators/local-setup.ts [hunk 3]
```diff
@@ -42,24 +43,58 @@ const mapServiceToNXname = async (serviceName: string) => {
...
+
+  // dump all env values to files
+  await Promise.all(
+    Object.entries(dockerComposeServices).map(async ([name, svc]) => {
+      const serviceNXName = await mapServiceToNXname(name)
+      await writeFile(
+        join(rootDir, `.env.${serviceNXName}`),
+        Object.entries(svc.env)
+          .map(
+            ([name, value]) => `export ${name}="${value.replace(/"/g, '\\"')}"`,
```
Comment (AndesKrrrrrrrrrrr) [inline infra/src/dsl/value-files-generators/local-setup.ts:91]: I think we need more handling here. Some values might contain backslashes, and even be multi-line json. Using `'` instead of `"` would be better. We don't know what escape characters might lure in some secrets.

**Review by AndesKrrrrrrrrrrr — COMMENTED** [review]
LGTM

Only edge cases left. Would be good to have a test case for multiline escape-character'd json secret.

File: infra/src/dsl/value-files-generators/local-setup.ts (hunk 3 above)
Comment (pshomov) [reply infra/src/dsl/value-files-generators/local-setup.ts]: Aaa, of course. Well spotted. Fixing it.

**Review by AndesKrrrrrrrrrrr — APPROVED** [review]
